### PR TITLE
Release/0.9.0

### PR DESCRIPTION
**Pull request by ilons**
## [0.9.0-rc1] - 2020-09-14
**This release requires Python 3.6 or later, and Terraform 0.12 or later.**  
**Be aware of embedded variable bugfix, which could break workarounds.**  
### Added
* Created ``terrascript.Terrascript.__iter__()`` method for iterating over resources, data sources, etc.  (issue #98).
* Make target and code to fail builds unless python code format passes Black
* Provider `google-beta`
* Release instruction for documentation
* RTD theme no longer bundled with Sphinx added as dev requirement
### Fixed
* Documentation generation, tests and examples cleaned up
* Missing dev requirement Black added
* Re-introduced ``terrascript.Terrascript.update()`` method (issue #98).
* String interpolation of variable should now properly result in a var.name reference (issue #109)
* terrascript module should now pass pep8 validation and have no lint errors
* Tool for generating providers now outputs code (almost) compliant with black
### Changed
* All Python code is now automatically formatted using [Black](https://pypi.org/project/black/).
* Code for providers updated to conform with black
* Contributors are now sorted in alphabetical order
* Documentation link from readme now points toward develop build on readthedocs.io to avoid 404
* Documentation on variable usage updated to reflect changes in issue #109
* Makefiles cleaned up, and help target contents now dynamically generated from comments
* Requirements files are now sorted alphabetically and duplicate entries are removed
* Terraform version supported updated from 0.12.15-19 to 0.12.29 + 0.13.0-2
* Updated code for providers generated
### Deleted
* Python 3.5 no longer supported as the official support are dropped: https://devguide.python.org/#status-of-python-branches
### Deprecated
* Individual modules for each provider (Example: ``terrascript.aws.r``) are now deprecated in favour of
  single modules as added in release 0.8.0.

### CODE REVIEW

**Comment by ilons** [comment]
@mjuenema I think this release does not add too much to worry about, and should be relatively safe.

**Comment by ilons** [comment]
@mjuenema When it comes to pypi, and readthedocs, is this automatically build and deployed by travis, or is there some manual step required when making a release?

**Comment by ilons** [comment]
Waiting to pull in https://github.com/mjuenema/python-terrascript/pull/119 as well

**Comment by mjuenema** [comment]
@ilons Check ``RELEASING.md``. I hope it's still up-to-date, otherwise correct the document as needed. That's what I used to do manually but maybe some steps could be automated.

**Review by mjuenema — APPROVED** [review]
I am happy!

**Comment by ilons** [comment]
@mjuenema I'll make an attempt to build and upload a RC for this.

**Comment by ilons** [comment]
@mjuenema We would either have to share the credentials for pypi, or set it up in the travis job to automatically build and push tags to pypi.
I would vote for the later, but that might not be done in time for 0.9.0 (on friday).

There is now a 0.9.0-rc1 tag which would be good if you could push to pypi test for me to test.

**Comment by mjuenema** [comment]
Let me check how that works with Travis-CI. If it's too difficult I am going to do it manually. Having it done automatically would be nice.

**Comment by ilons** [comment]
> Let me check how that works with Travis-CI. If it's too difficult I am going to do it manually. Having it done automatically would be nice.

Indeed would be nice. 
My suggestion is to build and publish all tags mathching `X.X.X-rcX` to pypi test, and tags matching `X.X.X` to pypi.

**Comment by mjuenema** [comment]
That sounds like a good approach. I should be able to set this up tomorrow morning, Australian time. That's late night Central European Time.

**Comment by mjuenema** [comment]
I have a ``.travis.yml`` file that (hopefully) makes Travis-CI automatically deploy to PyPi and TestPyPi. Getting the conditions right to publish all tags matchhing X.X.X-rcX to pypi test, and tags matching X.X.X to pypi is a bit tricky. Time will tell whether it works as expected. I successfully tested it, but only without those exact conditions.
What branch would you like me to create a Pull Request against?

**Comment by ilons** [comment]
> I have a `.travis.yml` file that (hopefully) makes Travis-CI automatically deploy to PyPi and TestPyPi. Getting the conditions right to publish all tags matchhing X.X.X-rcX to pypi test, and tags matching X.X.X to pypi is a bit tricky. Time will tell whether it works as expected. I successfully tested it, but only without those exact conditions.
> What branch would you like me to create a Pull Request against?

Make a PR against develop and we can look at it and discuss it there :+1:

File: .travis.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ deploy:
     distributions: "sdist bdist_wheel"
     on:
       all_branches: true
-      condition: $TRAVIS_TAG =~ \.rc[0-9]+$
+      condition: $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$
```
Comment (ilons) [inline .travis.yml:40]: @mjuenema I think this should simply match semver with no suffixes for tags that goes to live, instead of only rcx suffixes.

Comment (mjuenema) [reply .travis.yml:40]: Looks alright.

Comment (ilons) [reply .travis.yml:40]: Yeah, I think it does too. 
I don't know why Travis keep building the rc3 [over](https://www.travis-ci.org/github/mjuenema/python-terrascript/jobs/744825554#L3973) and [over](https://www.travis-ci.org/github/mjuenema/python-terrascript/jobs/744832993#L3973) and [over](https://www.travis-ci.org/github/mjuenema/python-terrascript/jobs/744902719#L3973) again, even though there are more [recent](https://github.com/mjuenema/python-terrascript/releases/tag/0.9.0rc4) [tags](https://github.com/mjuenema/python-terrascript/releases/tag/0.9.0rc5) [pushed](https://github.com/mjuenema/python-terrascript/releases/tag/0.9.0rc6).

Comment (ilons) [reply .travis.yml:40]: I was forgetting to update the docs/conf.py and (importantly) the terrascript/__init__.py, which was why..

**Pull request merged**